src/app/modules/main/chat_section/chat_content/input_area/controller.nim

```diff
@@ -229,11 +234,16 @@ proc loadLinkPreviews*(self: Controller, urls: seq[string]) =
     self.messageService.asyncUnfurlUrls(urls)
 
 proc setLinkPreviewEnabled*(self: Controller, enabled: bool) =
-  if(enabled):
+  if enabled and self.settingsService.saveUrlUnfurlingMode(UrlUnfurlingMode.Enabled):
     self.linkPreviewPersistentSetting = UrlUnfurlingMode.Enabled
     self.linkPreviewCurrentMessageSetting = UrlUnfurlingMode.Enabled
-  else:
+  elif not enabled and self.settingsService.saveUrlUnfurlingMode(UrlUnfurlingMode.Disabled):
     self.linkPreviewPersistentSetting = UrlUnfurlingMode.Disabled
     self.linkPreviewCurrentMessageSetting = UrlUnfurlingMode.Disabled
 
   self.delegate.setAskToEnableLinkPreview(false)
+
+proc onUnfurlingModeChanged(self: Controller, value: UrlUnfurlingMode) =
+  self.linkPreviewPersistentSetting = value
+  self.resetLinkPreviews()
+  self.setText(self.delegate.getText(), self.linkPreviewPersistentSetting == UrlUnfurlingMode.Enabled)
```

Comment:
Should we use `self.getLinkPreviewEnabled()` instead? 
It seems that this won't work for `AlwaysAsk` persistent setting.